src/common/mfu_io.c

```diff
@@ -99,9 +339,37 @@ int mfu_utimensat(int dirfd, const char *pathname, const struct timespec times[2
     return rc;
 }
 
-/* calls lstat, and retries a few times if we get EIO or EINTR */
-int mfu_lstat(const char* path, struct stat* buf)
-{
+int daos_stat(const char* path, struct stat* buf, mfu_file_t* mfu_file) {
+#ifdef DAOS_SUPPORT
+    char* name     = NULL;
+    char* dir_name = NULL;
+    parse_filename(path, &name, &dir_name);
+    assert(dir_name);
+
+    dfs_obj_t *parent = NULL;
+    int rc;
+    if (mfu_file->only_daos) {
+        rc = dfs_lookup(mfu_file->dfs, dir_name, O_RDWR, &parent, NULL, NULL);
+        if (rc) {
+            MFU_LOG(MFU_LOG_ERR, "dfs_lookup %s failed", dir_name);
+            errno = ENOENT;
+            rc = -1;
+        }
+    } else {
+        parent = lookup_insert_dir(dir_name, mfu_file);
+    }
+    rc = dfs_stat(mfu_file->dfs, parent, name, buf);
+    if (rc) {
+        MFU_LOG(MFU_LOG_ERR, "dfs_stat %s failed (%d %s)",
+                name, rc, strerror(rc));
+        errno = rc;
+        rc = -1;
+    }
```

Comment:
Should we avoid this call if the lookup for the parent fails?

Reply:
@adammoody well you can call it in the case that the lookup is on the dfs root (ie "/"), which doesn't have a parent.

Reply:
Let's add a comment about this, too, since it's another detail I'll probably forget :-)